src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -816,6 +818,70 @@ def _add_ingress_appgw_addon_role_assignment(result, cmd):
                                    'specified in %s addon. '
                                    'Are you an Owner on this subscription?', vnet_id, CONST_INGRESS_APPGW_ADDON_NAME)
 
+def aks_maintenanceconfiguration_list(cmd, # pylint: disable=too-many-locals,too-many-statements,too-many-branches
+    client,
+    resource_group_name,
+    resource_name
+):
+    return client.list_by_managed_cluster(resource_group_name, resource_name)
+
+def aks_maintenanceconfiguration_show(cmd, # pylint: disable=too-many-locals,too-many-statements,too-many-branches)
+    client,
+    resource_group_name,
+    resource_name,
+    config_name
+):
+    logger.warning('resource_group_name: %s, resource_name: %s, config_name: %s ', resource_group_name, resource_name, config_name)
+    return client.get(resource_group_name, resource_name, config_name)
+
+def aks_maintenanceconfiguration_delete(cmd, # pylint: disable=too-many-locals,too-many-statements,too-many-branches)
+    client,
+    resource_group_name,
+    resource_name,
+    config_name
+):
+    logger.warning('resource_group_name: %s, resource_name: %s, config_name: %s ', resource_group_name, resource_name, config_name)
+    return client.delete(resource_group_name, resource_name, config_name)
+
+def aks_maintenanceconfiguration_add(cmd, # pylint: disable=too-many-locals,too-many-statements,too-many-branches)
+    client,
+    resource_group_name,
+    resource_name,
+    config_name,
+    time_in_week,
+    not_allowed_time
+):
+    return aks_maintenanceconfiguration_update(cmd, client, resource_group_name, resource_name, config_name, time_in_week, not_allowed_time)
+
+def aks_maintenanceconfiguration_update(cmd, # pylint: disable=too-many-locals,too-many-statements,too-many-branches)
+    client,
+    resource_group_name,
+    resource_name,
+    config_name,
+    time_in_week,
+    not_allowed_time
+):
+    logger.warning('resource_group_name: %s, resource_name: %s, config_name: %s ', resource_group_name, resource_name, config_name)
+    logger.warning('time in week: %s, not_allowed_time: %s ', time_in_week, not_allowed_time)
+    week_schedule = []
+    if time_in_week != None:
+        week_schedule_dict = json.loads(time_in_week)
+        for item in week_schedule_dict:
+            w = TimeInWeek(**item)
```

Comment:
Nit: Python is much more verbose on naming than Go so normally you would call this week or some other full name instead of w. But not a big deal